mm2src/coins/nft.rs

```diff
@@ -444,18 +465,41 @@ async fn send_request_to_uri(uri: &str) -> MmResult<Json, GetInfoFromUriError> {
     Ok(response)
 }
 
-async fn try_get_uri_meta(token_uri: &Option<String>) -> MmResult<UriMeta, GetNftInfoError> {
-    match token_uri {
-        Some(token_uri) => {
-            if let Ok(response_meta) = send_request_to_uri(token_uri).await {
-                let uri_meta_res: UriMeta = serde_json::from_str(&response_meta.to_string())?;
-                Ok(uri_meta_res)
-            } else {
-                Ok(UriMeta::default())
+fn check_moralis_ipfs_bafy(token_uri: Option<&str>) -> Option<String> {
+    token_uri.map(|uri| {
+        if uri.contains("https://ipfs.moralis.io") && uri.contains("bafy") {
+            let parts: Vec<&str> = uri.splitn(2, "/ipfs/").collect();
+            format!("https://ipfs.io/ipfs/{}", parts[1])
+        } else {
+            uri.to_string()
+        }
+    })
```

Comment:
You can get around or change the result of this check if you provide incorrect information. The only thing this function does is it checks whether the string has specific values, but it doesn't check their order. To perform this check correctly, you should use the `Url` method.

Reply:
will add it later, to make it easier to review, as we have a lot of notes today

Reply:
done, now `check_moralis_ipfs_bafy` use Url to parse url from token_uri, but if it is invalid, we should leave it as it is. [related comment](https://github.com/KomodoPlatform/atomicDEX-API/pull/1877#discussion_r1244863204)